tests/test_checkFile.py

```diff
@@ -27,38 +27,98 @@ def test_no_file_exception():
     args.file = "resources/test.html"
 
 
-@mock.patch("src.checkFile.checkFile.headRequest")
-def test_headRequest_200(mock_headRequest):
+def mocked_requests_get(*args, **kwargs):
+    class MockResponse:
+        def __init__(self, status):
+            self.status = status
 
+        def status(self):
+            return self
+
+    if args[1] == "http://google.com":
+        return MockResponse(400)
+    elif args[1] == "http://google.cim":
+        return MockResponse(404)
+
+    return MockResponse("???")
+
+
+@mock.patch("urllib3.PoolManager.request", side_effect=mocked_requests_get)
+def test_headRequest_200(self):
     link = "http://google.com"
-    mock_headRequest.return_value = {"url": link, "status": 200, "secured": False}
     cF = checkFile.checkFile(args)
+    cF.headRequest(link)
 
-    assert cF.headRequest(link) == {
+    assert cF.allLinks[0] == {
         "url": "http://google.com",
-        "status": 200,
+        "status": 400,
         "secured": False,
     }
 
 
-@mock.patch("src.checkFile.checkFile.headRequest")
-def test_headRequest_404(mock_headRequest):
+@mock.patch("urllib3.PoolManager.request", side_effect=mocked_requests_get)
+def test_headRequest_404(self):
 
     link = "http://google.cim"
-    mock_headRequest.return_value = {"url": link, "status": 404, "secured": False}
     cF = checkFile.checkFile(args)
+    cF.headRequest(link)
 
-    assert cF.headRequest(link) == {
+    assert cF.allLinks[0] == {
         "url": "http://google.cim",
         "status": 404,
         "secured": False,
     }
 
 
+@mock.patch("urllib3.PoolManager.request", side_effect=mocked_requests_get)
+def test_headRequest_unknown(self):
+
+    link = ""
+    cF = checkFile.checkFile(args)
+    cF.headRequest(link)
+
+    assert cF.allLinks[0] == {
+        "url": "",
+        "status": "???",
+        "secured": False,
+    }
+
+
 def test_parseWebAddress():
 
     lineToParse = "https://www.google.com/search?q=help"
 
     cF = checkFile.checkFile(args)
 
-    assert cF.parseWebAddress(lineToParse) == "https://www.google.com/search?q=help"
\ No newline at end of file
+    assert cF.parseWebAddress(lineToParse) == "https://www.google.com/search?q=help"
+
+
+@mock.patch("urllib3.PoolManager.request", side_effect=mocked_requests_get)
+def test_printAll(self, capsys):
+    link = "http://google.com"
+    cF = checkFile.checkFile(args)
+    cF.headRequest(link)
+    cF.printAll()
+    captured = capsys.readouterr()
+    assert captured.out == "\x1b[91m[400] http://google.com\x1b[0m\n"
+
+
+@mock.patch("urllib3.PoolManager.request", side_effect=mocked_requests_get)
+def test_printBadResultsJSON(self, capsys):
+    link = "thisdoesnotwork"
+    cF = checkFile.checkFile(args)
+    cF.headRequest(link)
+    cF.printAll()
+    captured = capsys.readouterr()
+    assert captured.out == "\x1b[90m[???] thisdoesnotwork\x1b[0m\n"
+
+
+@mock.patch("urllib3.PoolManager.request", side_effect=mocked_requests_get)
+def test_checkThatFile(self, capsys):
```

Comment:
Same here. Could you change the title to better represent what's being tests
test_printAllJSON